pageserver/src/walreceiver/walreceiver_connection.rs

```diff
@@ -184,7 +184,20 @@ pub async fn handle_walreceiver_connection(
             replication_message = physical_stream.next() => replication_message,
         }
     } {
-        let replication_message = replication_message?;
+        let replication_message = match replication_message {
+            Ok(message) => message,
+            Err(replication_error) => {
+                if replication_error.is_closed() {
+                    info!("Replication stream got closed");
+                    return Ok(());
+                } else {
+                    return Err(anyhow::anyhow!(
+                        "Replication stream error: {replication_error}"
+                    ));
```

Comment:
By printing replication_error's `Display` value we lose a lot of detail here.
Is that intentional?
Why not use `.context()?`

Reply:
Direct `.context` invocations are not working on that error, because it does not implement `Display`, but I've wrapped it differently, should be better now.